nemo/lightning/megatron_parallel.py

```diff
@@ -489,6 +503,41 @@ def infer_num_microbatches(self, data: Union[DataT, Iterator[DataT], List[Iterat
 
         raise ValueError("Cannot infer `num_microbatches` from data, please specify it manually")
 
+    def init_tp_comm_overlap(self, micro_batch_size, sequence_length):
```

Comment:
The performance optimizations are all self contained in the `MegatronCommOverlap` plugin now. The only exception is this function which needs to run to initialize some buffers for TP overlap
This function needs to know the microbatch size, but I'm not how we can nicely get that while inside a callback @marcromeyn @gautham-kollu do you have any ideas?

Reply:
Nevermind...I learned about `megatron.core.num_microbatches_calculator` so I can probably use that